stan/math/prim/mat/fun/gp_matern52_cov.hpp

```diff
@@ -0,0 +1,302 @@
+    check_not_nan("gp_matern52_cov", "x1", x2[n]);
+
+  check_positive_finite("gp_matern52_cov", "marginal variance", sigma);
+  check_positive_finite("gp_matern52_cov", "length scale", length_scale);
+
+  Eigen::Matrix<typename return_type<T_x1, T_x2, T_s, T_l>::type,
+                Eigen::Dynamic, Eigen::Dynamic>
+      cov(x1_size, x2_size);
+
+  if (x1_size == 0 || x2_size == 0)
+    return cov;
+
+  T_s sigma_sq = square(sigma);
+  T_l root_5_inv_l = sqrt(5.0) / length_scale;
+  T_l inv_l_sq_5_3 = 5.0 / (3.0 * square(length_scale));
+  T_l neg_root_5_inv_l = -sqrt(5.0) / length_scale;
+
+  for (size_t i = 0; i < x1_size; ++i) {
+    for (size_t j = 0; j < x2_size; ++j) {
+      typename return_type<T_x1, T_x2>::type sq_distance
+          = squared_distance(x1[i], x2[j]);
+      typename return_type<T_x1, T_x2>::type root_sq_distance
+          = sqrt(sq_distance);
+      cov(i, j) = sigma_sq
+                  * (1.0 + root_5_inv_l * root_sq_distance
+                     + inv_l_sq_5_3 * sq_distance)
+                  * exp(neg_root_5_inv_l * root_sq_distance);
+    }
+  }
+  return cov;
+}
+
+/**
+ * Returns a Matern 5/2 covariance matrix with two input vectors
+ * with automatic relevance determination (ARD).
+ *
+ * \f[ k(x, x') = \sigma^2\bigg(1 +
+ *  \sqrt{5}\sqrt{\sum_{k=1}^{K}\frac{d(x, x')^2}{l_k^2}} +
+ *  \frac{5}{3} \sqrt{\sum_{k=1}^{K}\frac{d(x, x')^2}{l_k^2}}\bigg)
+ *  exp\bigg(-\frac{5}{3}\bigg(\sqrt{\sum_{k=1}^K{\frac{d(x, x')^2}{l_k^2}}
+ *  }\bigg)\bigg)
+ * \f]
+ *
+ * where \f$ d(x, x') \f$ is the Euclidean distance.
+ *
+ * @tparam T_x1 type of elements contained in vector x1
+ * @tparam T_x2 type of elements contained in vector x2
+ * @tparam T_s type of element of sigma, marginal standard deviation
+ * @tparam T_l type of elements in vector of length scale
+ *
+ * @param x1 std::vector of elements that can be used in stan::math::distance
+ * @param x2 std::vector of elements that can be used in stan::math::distance
+ * @param length_scale length scale
+ * @param sigma standard deviation that can be used in stan::math::square
+ * @throw std::domain error if sigma <= 0, l <= 0, or x is nan or inf
+ * @throw std::invalid_argument if length scale size != dimension of x1 or x2
+ *
+ */
+template <typename T_x1, typename T_x2, typename T_s, typename T_l>
+inline typename Eigen::Matrix<
+    typename stan::return_type<T_x1, T_x2, T_s, T_l>::type, Eigen::Dynamic,
+    Eigen::Dynamic>
+gp_matern52_cov(const std::vector<T_x1> &x1, const std::vector<T_x2> &x2,
+                const T_s &sigma, const std::vector<T_l> &length_scale) {
+  using std::exp;
+
+  size_t x1_size = x1.size();
+  size_t x2_size = x2.size();
+  size_t l_size = length_scale.size();
+
+  for (size_t n = 0; n < x1_size; ++n)
+    check_not_nan("gp_matern52_cov", "x1", x1[n]);
+  for (size_t n = 0; n < x2_size; ++n)
+    check_not_nan("gp_matern52_cov", "x1", x2[n]);
+
+  check_positive_finite("gp_matern52_cov", "marginal variance", sigma);
```

Comment:
I think this is standard deviation. Please verify and make everything consistent.

Reply:
I'm just using magnitude for these from now on, as this is consistent with GP literature